services/user/Permissions/ui/src/permissions/App.tsx

```diff
@@ -1,83 +1,110 @@
 import { useEffect, useState } from "react";
 
 import { Button } from "@shadcn/button";
-import { Checkbox } from "@shadcn/checkbox";
-
-import { siblingUrl } from "@psibase/common-lib";
-import { Nav } from "@components/nav";
 
 import { supervisor } from "./perms_main";
+import { siblingUrl } from "@psibase/common-lib";
+import { ActivePermsOauthRequest } from "./db";
 
 export const App = () => {
     const thisServiceName = "permissions";
-    const [isLoading, setIsLoading] = useState<boolean>(true);
-    const [params, setParams] = useState<any>({});
-    const [isValidPermRequest, setIsValidPermRequest] = useState<any>({});
-
-    const init = async () => {
+    const [isLoading, setIsLoading] = useState(true);
+    const [validPermRequest, setValidPermRequest] = useState<any>(null);
+    const [error, setError] = useState<string | null>(null);
+    const initApp = async () => {
         await supervisor.onLoaded();
```

Comment:
Just an FYI: Instead of exporting this from `perms_main` and importing it into this file, you can call `const supervisor = getSupervisor();` in this file and you'll get the same instance now every time.